### PR TITLE
Add functionality to get recommendations based on user's watch lists

### PR DESCRIPTION
**Pull request by Caruki**
Add functions to calculate and get recommendations based on the user's watch lists by genres and networks

You can try the live Preview with your own account or this example account:
https://show-cased-p-get-recs-4fnfvkzz.herokuapp.com/recs
email: hallo@hallo.com
pw: hallihallo
(however there is a bug with lists (no display yet) and on newest page.. these are fixed in another PR 🙈 )

### CODE REVIEW

File: lib/models/shows.js [hunk 1]
```diff
@@ -65,13 +63,16 @@ async function getShowDetails(showId) {
...
     poster: `https://image.tmdb.org/t/p/w780${show.images.backdrops[0].file_path}`,
     rating: show.vote_average,
-    genres: show.genres.map((genre) => {
+    genres_id: show.genres.map((genre) => {
+      return genre.id;
+    }),
+    genres_name: show.genres.map((genre) => {
       return genre.name;
     }),
+    networks: show.networks,
```
Comment (lmachens) [inline lib/models/shows.js:75]: The naming convention is different to your other variables:

```suggestion
    genreIds: show.genres.map((genre) => {
      return genre.id;
    }),
    genreNames: show.genres.map((genre) => {
      return genre.name;
    }),
    networks: show.networks,
```

File: lib/models/recs.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+const User = require('../models/User');
+const fetch = require('node-fetch');
+
+function calculateTopThree(dataArray1, dataArray2) {
```
Comment (lmachens) [inline lib/models/recs.js:4]: `dataArray1` and `dataArray2`? 
The naming is too vague.

File: lib/models/recs.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  const dataArray1Reduced = dataArray1.reduce(function (array, element) {
+    return array.concat(element);
+  });
```
Comment (lmachens) [inline lib/models/recs.js:7]: What do you try to do here?

File: lib/models/recs.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  const user = await User.findById(userId);
+  if (!user) {
+    return 'User not found';
+  } else if (!user.towatch || !user.watched) {
+    return null;
+  } else {
+    const toWatchListAllGenres = user.towatch.map((show) => {
+      const genres = show.genres_id.map((genre) => {
+        return genre;
+      });
```
Comment (lmachens) [inline lib/models/recs.js:41]: This `.map` doesn't make sense 😃. Input and output are equal.

File: lib/models/recs.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+async function getRecsFromAPI(data, userId, filter) {
+  const response = await fetch(
+    ` https://api.themoviedb.org/3/discover/tv?api_key=${
+      process.env.TMDB_API_KEY
+    }&language=en-US&sort_by=popularity.desc&page=1&timezone=America%2FNew_York&with_${filter}=${
+      data[0]
+    }${filter === 'genres' ? '%2C' : '%7C'}${data[1]}${
+      filter === 'genres' ? '%2C' : '%7C'
+    }${data[2]}&include_null_first_air_dates=false`
+  );
```
Comment (lmachens) [inline lib/models/recs.js:115]: What about this part? `filter === 'genres' ? '%2C' : '%7C'`
You can create a new variable for this to make it easier to understand.

File: lib/models/recs.js (hunk 2 above)
Comment (Caruki) [reply lib/models/recs.js:4]: I thought I would name it unspecific so I can reuse the function. I changed it to different names specific to this project now

File: lib/models/recs.js (hunk 4 above)
Comment (Caruki) [reply lib/models/recs.js:41]: yeah this was a leftover from a different approach before. Thanks for the heads up 👍

File: lib/models/recs.js (hunk 3 above)
Comment (Caruki) [reply lib/models/recs.js:7]: the data arrays this function receives have nested arrays inside.  I am using concat to turn them all into one normal array

**Pull request merged**